nodeup/pkg/model/crictl.go

```diff
@@ -0,0 +1,48 @@
+package model
+
+import (
+	"fmt"
+
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+	"k8s.io/kops/util/pkg/distributions"
+)
+
+type CrictlBuilder struct {
+	*NodeupModelContext
+}
+
+var _ fi.NodeupModelBuilder = &CrictlBuilder{}
+
+func (b *CrictlBuilder) Build(c *fi.NodeupModelBuilderContext) error {
+	assetName := "crictl"
+	assetPath := ""
+	asset, err := b.Assets.Find(assetName, assetPath)
+	if err != nil {
+		return fmt.Errorf("unable to locate asset %q", err)
+	}
+
+	c.AddTask(&nodetasks.File{
+		Path:     b.crictlPath(),
+		Contents: asset,
+		Type:     nodetasks.FileType_File,
+		Mode:     s("0755"),
+	})
+
+	return nil
+}
+
+func (b *CrictlBuilder) binaryPath() string {
+	path := "/usr/local/bin"
+	if b.Distribution == distributions.DistributionFlatcar {
+		path = "/opt/kops/bin"
+	}
+	if b.Distribution == distributions.DistributionContainerOS {
+		path = "/home/kubernetes/bin"
+	}
+	return path
+}
+
+func (b *CrictlBuilder) crictlPath() string {
+	return b.binaryPath() + "/crictl"
```

Comment:
Nit: You should be using filepath.Join or path.Join to avoid hard-coding the path separator.  (And in most cases it's filepath.Join for files, path.Join for files).  In this case though it really doesn't matter because we don't support windows and presumably it would be crictl.exe anyway!